daemon/UnixDaemon.cpp

```diff
@@ -171,19 +180,27 @@ namespace i2p
 			}
 			gracefulShutdownInterval = 0; // not specified
 
-			// Signal handler
-			struct sigaction sa;
-			sa.sa_handler = handle_signal;
-			sigemptyset(&sa.sa_mask);
-			sa.sa_flags = SA_RESTART;
-			sigaction(SIGHUP, &sa, 0);
-			sigaction(SIGUSR1, &sa, 0);
-			sigaction(SIGABRT, &sa, 0);
-			sigaction(SIGTERM, &sa, 0);
-			sigaction(SIGINT, &sa, 0);
-			sigaction(SIGPIPE, &sa, 0);
-
-			return Daemon_Singleton::start();
+            // handle signal TSTP
+            bool handleTSTP; i2p::config::GetOption("unix.handle_sigtstp", handleTSTP);
+
+            // Signal handler
+            struct sigaction sa;
+            sa.sa_handler = handle_signal;
+            sigemptyset(&sa.sa_mask);
+            sa.sa_flags = SA_RESTART;
+            sigaction(SIGHUP, &sa, 0);
+            sigaction(SIGUSR1, &sa, 0);
+            sigaction(SIGABRT, &sa, 0);
+            sigaction(SIGTERM, &sa, 0);
+            sigaction(SIGINT, &sa, 0);
+            sigaction(SIGPIPE, &sa, 0);
+            if (handleTSTP)
+            {
+                sigaction(SIGTSTP, &sa, 0);
+                sigaction(SIGCONT, &sa, 0);
+            }
```

Comment:
I recommend replace 0 to nullptr, this is more readable:
```suggestion
            sigaction(SIGHUP, &sa, nullptr);
            sigaction(SIGUSR1, &sa, nullptr);
            sigaction(SIGABRT, &sa, nullptr);
            sigaction(SIGTERM, &sa, nullptr);
            sigaction(SIGINT, &sa, nullptr);
            sigaction(SIGPIPE, &sa, nullptr);
            if (handleTSTP)
            {
                sigaction(SIGTSTP, &sa, nullptr);
                sigaction(SIGCONT, &sa, nullptr);
            }
```